### ISSUE TITLE
Postgres: add support for GENERATED ALWAYS AS (...) VIRTUAL

### ISSUE THREAD

**Issue by TeeAaTeeUu** [sqlfluff/sqlfluff#8139]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.


### What Happened

[PostgreSQL18](https://www.postgresql.org/docs/current/sql-createforeigntable.html) has added virtual [generated columns](https://www.postgresql.org/docs/current/ddl-generated-columns.html).

`VIRTUAL` seems to be [unreserved / non-reserved](https://github.com/postgres/postgres/blame/1ae87df63ff693a91fbcb968850103a5a5d8cd96/src/backend/parser/gram.y#L19246). Also `STORED` isn't anymore required, and defaults to `VIRTUAL` if both are omitted.

Does `sqlfluff` have support for restricting this only for Postgres v18, and still failing on <= v17?

### Expected Behaviour

```bash
> sqlfluff lint --dialect postgres test.sql
All Finished 📜 🎉!  
```

### Observed Behaviour

```bash
> sqlfluff lint --dialect postgres test.sql

== [../test.sql] FAIL                                                                                                                                                                                                                                             
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'CREATE
                       | TABLE test (\n  my_number SMALLINT...'
```

### How to reproduce

```sql
CREATE TABLE test (
  my_number SMALLINT NOT NULL DEFAULT 42,
  is_small BOOLEAN GENERATED ALWAYS AS (
    my_number < 22
  ) VIRTUAL
)
```

### Dialect

Postgres, v18

### Version

`sqlfluff, version 4.2.2`
`Python 3.14.6`

### Configuration

```
[sqlfluff]
dialect = postgres
```

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by TeeAaTeeUu** [comment]
To my knowledge, only following changes are needed (_in addition to tests_):

<details><summary>git --diff</summary>

```diff
diff --git a/src/sqlfluff/dialects/dialect_postgres.py b/src/sqlfluff/dialects/dialect_postgres.py
index 4e4d0c512..2682668b7 100644
--- a/src/sqlfluff/dialects/dialect_postgres.py
+++ b/src/sqlfluff/dialects/dialect_postgres.py
@@ -3971,7 +3971,11 @@ class ColumnConstraintSegment(ansi.ColumnConstraintSegment):
                     Ref("ExpressionSegment"),
                 ),
             ),
-            Sequence("GENERATED", "ALWAYS", "AS", Ref("ExpressionSegment"), "STORED"),
+            Sequence(
+                "GENERATED", "ALWAYS", "AS",
+                Ref("ExpressionSegment"),
+                OneOf("STORED", "VIRTUAL", optional=True)
+            ),
             Sequence(
                 "GENERATED",
                 OneOf("ALWAYS", Sequence("BY", "DEFAULT")),
@@ -4056,8 +4060,12 @@ class ForeignTableColumnConstraintSegment(ansi.ColumnConstraintSegment):
                     Ref("ExpressionSegment"),
                 ),
             ),
-            # GENERATED ALWAYS AS ( generation_expr ) STORED
-            Sequence("GENERATED", "ALWAYS", "AS", Ref("ExpressionSegment"), "STORED"),
+            # GENERATED ALWAYS AS ( generation_expr ) [ STORED | VIRTUAL ]
+            Sequence(
+                "GENERATED", "ALWAYS", "AS",
+                Ref("ExpressionSegment"),
+                OneOf("STORED", "VIRTUAL", optional=True)
+            ),
         ),
     )
 
diff --git a/src/sqlfluff/dialects/dialect_postgres_keywords.py b/src/sqlfluff/dialects/dialect_postgres_keywords.py
index a977eac6b..e163aa973 100644
--- a/src/sqlfluff/dialects/dialect_postgres_keywords.py
+++ b/src/sqlfluff/dialects/dialect_postgres_keywords.py
@@ -890,6 +890,7 @@ postgres_docs_keywords = [
     ("VERSIONING", "not-keyword"),
     ("VIEW", "non-reserved"),
     ("VIEWS", "non-reserved"),
+    ("VIRTUAL", "non-reserved"),
     ("VOLATILE", "non-reserved"),
     ("WHEN", "reserved"),
     ("WHENEVER", "not-keyword"),
```

</details> 

Matching what e.g. [sqlfluff/dialects/dialect_sqlite.py](https://github.com/sqlfluff/sqlfluff/blob/main/src/sqlfluff/dialects/dialect_sqlite.py) and [sqlfluff/dialects/dialect_sqlite.py](https://github.com/sqlfluff/sqlfluff/blob/main/src/sqlfluff/dialects/dialect_mariadb.py) already have.
